File: kkn_backend/app/models/wali_nagari.go
package models

import "github.com/goravel/framework/database/orm"

type WaliNagari struct {
	orm.Model

	// Fields from your migration
	Name  string `orm:"column:name"`
	Email string `orm:"column:email"`
	Phone string `orm:"column:phone"`

	// Foreign key to User model
	UserId uint64 `orm:"column:user_id"`
	User   *User  `orm:"belongsTo:user"` // Relationship with User model

	// Timestamps are handled automatically by `orm.Model`
}

File: backend/app/Models/WaliNagari.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class WaliNagari extends Model
{
    protected $fillable = [
        'name', 'email', 'phone', 'user_id'
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function nagaris(): HasMany
    {
        return $this->hasMany(Nagari::class, 'wali_nagari_id');
    }
}
